src/twosection.jl

```diff
@@ -0,0 +1,85 @@
+"""
+    TwoSectionView{T<:Real} <: AbstractGraph{Int64}
+
+Create a 2-section view of a hypergraph `h`.
+Note this is a view - changes to the original hypergraph will be automatically reflected in the view.
+
+The 2-section view of a hypergraph is suitable for processing with the LightGraphs.jl package.
+Several LightGraphs methods are provided for the compability.
+
+"""
+struct TwoSectionView{T<:Real} <: AbstractGraph{Int}
+    h::Hypergraph{T}
+    function TwoSectionView{T}(h::Hypergraph{T}) where {T<:Real}
+        new(h)
+    end
+end
+
+LightGraphs.nv(t::TwoSectionView) = length(t.h.v2he)
+
+LightGraphs.vertices(t::TwoSectionView) = Base.OneTo(nv(t))
+
+LightGraphs.ne(t::TwoSectionView)::Int64 = 0.5*sum(length.(t.h.he2v) .* (length.(t.h.he2v) .- 1))
+
+"""
+    LightGraphs.all_neighbors(t::TwoSectionView, v::Integer)
+
+Returns N(v) (the vertex v is not included in N(v))
+"""
+function LightGraphs.all_neighbors(t::TwoSectionView, v::Integer)
+    neighbors = Set()
+    for he in keys(t.h.v2he[v])
+        union!(neighbors, keys(t.h.he2v[he]))
+    end
+    delete!(neighbors, v) #remove v from its neighborhood
+    convert(Array{Int64}, collect(neighbors)) #returns the corresponding array
```

Comment:
no need to `convert` I think, `collect` should return a vector.

Reply:
Yes and note that `LightGraphs` uses `Int` rather than `Int64`. The difference is that `Int` represents system native primitives (`Int` is 32-bit on 32-bit platforms and 64-bit on 64-bit platforms). Hence we should also use `Int`s rather than `Int64`

Reply:
That is why, in particular, I have recommended to remove the `convert` altogether.

Reply:
Without using `convert`, `all_neighbors` returns `Array{Any}`. Is it okay to have `Any` or do we need to force it to return `Array{Int}` (as for the BipartiteView)?

Reply:
The fix is that you should define above ` neighbors = Set{Int}()` (which creates set of integers) not `neighbors = Set()` (which creates set of `Any`).